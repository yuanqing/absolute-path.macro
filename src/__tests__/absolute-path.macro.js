const path = require('path')
const pluginTester = require('babel-plugin-tester')
const plugin = require('babel-plugin-macros')

const projectDirectoryAbsolutePath = path.join(__dirname, '..', '..')

expect.addSnapshotSerializer({
  test: function (value) {
    return typeof value === 'string'
  },
  print: function (value) {
    return value.replace(projectDirectoryAbsolutePath, '/<PROJECT_ROOT>')
  }
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename
  },
  tests: {
    'No usage': {
      code: `const { absolutePath } = require('../absolute-path.macro')`
    },
    'Incorrect usage': {
      code: `
        const { absolutePath } = require('../absolute-path.macro')
        console.log(absolutePath('foo'))
      `,
      error: true
    },
    'Empty template literal': {
      code: `
        const { absolutePath } = require('../absolute-path.macro')
        console.log(absolutePath\`\`)
      `
    },
    'Non-empty template literal': {
      code: `
        const { absolutePath } = require('../absolute-path.macro')
        console.log(absolutePath\`foo\`)
      `
    }
  }
})
