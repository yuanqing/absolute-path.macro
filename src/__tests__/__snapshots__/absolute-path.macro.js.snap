// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros Empty template literal: Empty template literal 1`] = `

const { absolutePath } = require('../absolute-path.macro')
console.log(absolutePath\`\`)

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(\`/<PROJECT_ROOT>/src/__tests__\`);

`;

exports[`macros Incorrect usage: Incorrect usage 1`] = `

const { absolutePath } = require('../absolute-path.macro')
console.log(absolutePath('foo'))

      ↓ ↓ ↓ ↓ ↓ ↓

Error: ../absolute-path.macro: Usage as Call Expression is not supported

`;

exports[`macros No usage: No usage 1`] = `

const { absolutePath } = require('../absolute-path.macro')

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros Non-empty template literal: Non-empty template literal 1`] = `

const { absolutePath } = require('../absolute-path.macro')
console.log(absolutePath\`foo\`)

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(\`/<PROJECT_ROOT>/src/__tests__/foo\`);

`;
